From b7d04be12e4fdc5180702937d1bcc9a167032f52 Mon Sep 17 00:00:00 2001
From: Will Ruggiano <wmruggiano@gmail.com>
Date: Thu, 16 Dec 2021 14:17:49 -0700
Subject: [PATCH] feat: pass QUTE_TAB to userscripts

---
 qutebrowser/browser/commands.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/qutebrowser/browser/commands.py b/qutebrowser/browser/commands.py
index 395d8e8a4..12ad9d575 100644
--- a/qutebrowser/browser/commands.py
+++ b/qutebrowser/browser/commands.py
@@ -1151,6 +1151,8 @@ class CommandDispatcher:
             env['QUTE_COUNT'] = str(count)

         idx = self._current_index()
+        env['QUTE_TAB'] = str(idx + 1)
+
         if idx != -1:
             env['QUTE_TITLE'] = self._tabbed_browser.widget.page_title(idx)

--
2.33.1
