diff --git a/qutebrowser/browser/commands.py b/qutebrowser/browser/commands.py
index 395d8e8a435..11f46b3b7ff 100644
--- a/qutebrowser/browser/commands.py
+++ b/qutebrowser/browser/commands.py
@@ -1151,6 +1151,9 @@ def _run_userscript(self, selection, cmd, args, verbose, output_messages,
             env['QUTE_COUNT'] = str(count)

         idx = self._current_index()
+        if idx != -1:
+            env['QUTE_TAB_INDEX'] = str(idx + 1)
+
         if idx != -1:
             env['QUTE_TITLE'] = self._tabbed_browser.widget.page_title(idx)
