diff --git a/Marksman/Marksman.fsproj b/Marksman/Marksman.fsproj
index 7b3ce9b..abaa1f4 100644
--- a/Marksman/Marksman.fsproj
+++ b/Marksman/Marksman.fsproj
@@ -8,12 +8,9 @@
         <EnableLint>false</EnableLint>
     </PropertyGroup>
     <Target Name="Version" BeforeTargets="BeforeBuild">
-        <Exec Command="git describe --always --dirty" ConsoleToMSBuild="true">
-            <Output TaskParameter="ConsoleOutput" PropertyName="VersionString"/>
-        </Exec>
         <PropertyGroup>
-            <VersionSuffix>$(VersionString)</VersionSuffix>
-            <Version>$(VersionPrefix)-$(VersionSuffix)</Version>
+            <VersionSuffix></VersionSuffix>
+            <Version>$(VersionPrefix)</Version>
         </PropertyGroup>
     </Target>
     <Target Name="FSharpLint" AfterTargets="BeforeBuild" Condition="$(EnableLint)">
